UnitTests/ApplicationTests.cs

```diff
@@ -120,31 +253,63 @@ public void Begin_End_Cleana_Up ()
 		}
 
 		[Fact]
-		public void RequestStop_Stops ()
+		public void Run_T_InitWithDriver_Works_with_TestTopLevel ()
 		{
 			// Setup Mock driver
 			Init ();
 
-			var top = new Toplevel ();
-			var rs = Application.Begin (top);
-			Assert.NotNull (rs);
-			Assert.Equal (top, Application.Current);
+			Application.Iteration = () => {
+				Application.RequestStop ();
+			};
+			
+			// Run<TestToplevel> when already initialized with a Driver will work
+			Application.Run<TestToplevel> (errorHandler: null);
+
+			Shutdown ();
+
+			Assert.Null (Application.Top);
+			Assert.Null (Application.MainLoop);
+			Assert.Null (Application.Driver);
+		}
 
+		[Fact]
+		public void Run_T_NoInit_ThrowsInDefaultDriver ()
```

Comment:
This test will fail on `Linux` because the `CursesDriver` will not throw any exception and stay waiting for some keystroke. Maybe checking the operation system. Or maybe you have to review the `Run<T>`.

Reply:
I had a feeling... I should have run in WSL. My bad.